src/fmpz/sqr.c

```diff
@@ -0,0 +1,70 @@
+/*
+    Copyright (C) 2024 Albin Ahlbäck
+
+    This file is part of FLINT.
+
+    FLINT is free software: you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.  See <https://www.gnu.org/licenses/>.
+*/
+
+#include "mpn_extras.h"
+#include "fmpz.h"
+
+__GMP_DECLSPEC extern void * (*__gmp_allocate_func)(size_t);
```

Comment:
I think we assume that GMP allocation functions are the same as FLINT's, so there shouldn't need to be some special magic in this file.

Reply:
Do we?

Reply:
I suspect there are already cases where we put a ``flint_malloc``:ed pointer in an ``mpz``. I could be wrong about this though.

Reply:
Then why don't we simply define `flint_malloc` as `__gmp_allocate_func`?